oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleConstants.java

```diff
@@ -57,9 +60,45 @@ public static String getConnectionString(String connectionType,
     }
     if (OracleConstants.SERVICE_CONNECTION_TYPE.equalsIgnoreCase(connectionType)) {
       return String.format(OracleConstants.ORACLE_CONNECTION_STRING_SERVICE_NAME_FORMAT,
-          host, port, database);
+             DEFAULT_CONNECTION_PROTOCOL, host, port, database);
     }
     return String.format(OracleConstants.ORACLE_CONNECTION_STRING_SID_FORMAT,
-        host, port, database);
+           DEFAULT_CONNECTION_PROTOCOL, host, port, database);
+  }
+
+  /**
+   * Constructs the Oracle connection string based on the provided connection type, host, port, and database.
+   * If SSL is enabled, the connection protocol will be "tcps" instead of "tcp".
+   *
+   * @param connectionType TNS/Service/SID
+   * @param host Host name of the oracle server
+   * @param port Port of the oracle server
+   * @param database Database to connect to
+   * @param useSSL Whether SSL/TLS is required(YES/NO)
+   * @return Connection String based on the given parameters and connection type.
+   */
+  public static String getConnectionString(String connectionType,
+                                           @Nullable String host,
+                                           @Nullable int port,
+                                           String database,
+                                           @Nullable String useSSL) {
+    // Use protocol as "tcps" when SSL is requested or else use "tcp".
+    String connectionProtocol;
+    if (useSSL != null && useSSL.equalsIgnoreCase("yes")) {
+      connectionProtocol = "tcps";
+    } else {
+      connectionProtocol = "tcp";
+    }
+
+    // For TNS Descriptor we can't specify the protocol in connection string.
+    if (OracleConstants.TNS_CONNECTION_TYPE.equalsIgnoreCase(connectionType)) {
```

Comment:
see if we can convert it into `swtich` statement as in general either of the statements will be true. So, we can avoid the compiler navigation to multiple conditional statements.

Reply:
Converted to switch [6c60d2d7a7a55d2af722dae6f8657449ada3efc1]